lib/roby/execution_engine.rb

```diff
@@ -2458,6 +2458,7 @@ def cycle_end(stats, raise_framework_errors: Roby.app.abort_on_application_excep
         # the beginning of the event loop, in propagation context. If the block
         # raises, the exception is raised back in the calling thread.
         def execute(catch: [], type: :external_events)
+            raise "a block is required" unless block_given?
```

Comment:
should be an ArgumentError

Reply:
I didn't understand

Reply:
The error raised should be `ArgumentError`, not `RuntimeError` (which is the default when you call `raise` with just a message.